File: BeltmanSoftwareDesign.Angular/BeltmanSoftwareDesign.Shared/Jsons/Workorder.cs
namespace BeltmanSoftwareDesign.Shared.Jsons
{
    public class Workorder
    {
        public long id { get; set; }

        public string? Name { get; set; }
        public string? Description { get; set; }
        public DateTime Start { get; set; }
        public DateTime Stop { get; set; }

        public long? ProjectId { get; set; }
        public string? ProjectName { get; set; }

        public long? CustomerId { get; set; }
        public string? CustomerName { get; set; }

        public List<InvoiceWorkorder>? InvoiceWorkorders { get; set; }
        public List<WorkorderAttachment>? WorkorderAttachments { get; set; }

        public double AmountUur
        {
            get
            {
                return (Stop - Start).TotalHours;
            }
            set
            {
                Stop = Start.AddHours(value);
            }
        }
    }
}

File: BeltmanSoftwareDesign.Angular/BeltmanSoftwareDesign.Angular/src/app/interfaces/workorder.ts
import { InvoiceWorkorder } from "./invoiceworkorder";
import { WorkorderAttachment } from "./workorderattachment";

export interface Workorder {
    id: number;
    name: string | null;
    description: string | null;
    start: Date;
    stop: Date;
    projectId: number | null;
    projectName: string | null;
    customerId: number | null;
    customerName: string | null;
    invoiceWorkorders: InvoiceWorkorder[];
    workorderAttachments: WorkorderAttachment[];
    amountUur: number;
}